test/click.spec.js

```diff
@@ -405,4 +405,41 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
       expect(await page.evaluate('document.querySelector("#button").style.left')).toBe('0px');
     });
   });
+
+  describe('Page.check', function() {
+    it('should check the box', async({page}) => {
+      await page.setContent(`<input id='checkbox' type='checkbox'></input>`);
+      await page.check('input');
+      expect(await page.evaluate(() => checkbox.checked)).toBe(true);
+    });
+    it('should not check the checked box', async({page}) => {
+      await page.setContent(`<input id='checkbox' type='checkbox' checked></input>`);
+      await page.check('input');
+      expect(await page.evaluate(() => checkbox.checked)).toBe(true);
+    });
+    it('should uncheck the box', async({page}) => {
+      await page.setContent(`<input id='checkbox' type='checkbox' checked></input>`);
+      await page.uncheck('input');
+      expect(await page.evaluate(() => checkbox.checked)).toBe(false);
+    });
+    it('should not uncheck the unchecked box', async({page}) => {
+      await page.setContent(`<input id='checkbox' type='checkbox'></input>`);
+      await page.uncheck('input');
+      expect(await page.evaluate(() => checkbox.checked)).toBe(false);
+    });
+    it('should check the box by label', async({page}) => {
+      await page.setContent(`<label for='checkbox'><input id='checkbox' type='checkbox'></input></label>`);
```

Comment:
Could you please add the following tests?

Checkbox not inside the label:
`<label for='checkbox'>Text</label><div><input id='checkbox' type='checkbox'></input></div>`

Checkbox inside a label without for=.
`<label>Text<span><input id='checkbox' type='checkbox'></input></span></label>`

Reply:
Done

Reply:
Done